src/authority/mzp.js

```diff
@@ -0,0 +1,169 @@
+/* Modifies Airtable connector to enable custom URI in TEI Publisher */
+
+import '@lrnwebcomponents/es-global-bridge';
+import { Registry } from './registry.js';
+import { resolveURL } from '../utils.js';
+
+function expandTemplate(template = '', options) {
+  return template.replace(/\${([^}]+)}/g, (match, p) => {
+      let replacement;
+      if (options[p]) {
+          replacement = options[p];
+      }
+      return replacement || '';
+  });
+}
+
+function getTemplate(configElem, selector) {
+  const template = configElem.querySelector(selector);
+  if (template instanceof HTMLTemplateElement) {
+    const wrapper = document.createElement('div');
+    wrapper.appendChild(template.content.cloneNode(true));
+    return wrapper.innerHTML;
+  }
+  return '';
+}
+
+export class Airtable extends Registry {
```

Comment:
Shouldn't this class be named differently to avoid conflicts with the existing `Airtable` class?

Reply:
I think we could streamline this by making the ID to use configurable. I'll make a proposal tomorrow...

Reply:
So I would propose the following: instead of creating a new connector, change the existing airtable connector to accept an additional attribute `use-as-key` in the config element. Right now I can't test this myself, but I'd say the required changes would be:

1. in line 35 add another variable `this.idField = configElem.getAttribute('use-as-key');`
2. in line 101, change the `id` property to `id: this.idField ? record.get(this.idField) : record.id,`
3. in line 120, we also need to take into account that the `filterByFormula` expression needs to be different, so replace with `filterByFormula: this.idField ? `${this.idField}='${key}'` : `RECORD_ID()='${key}'`
4. then again in line 160 change the `id` property: `id: this.idField ? record.get(this.idField) : record.id,`

Could you try this and report if it works?